packages/mdc-textfield/_mixins.scss

```diff
@@ -528,350 +950,468 @@
   @include mdc-text-field-icon-horizontal-position-two-icons_(
     $mdc-text-field-icon-position,
-    $mdc-text-field-trailing-icon-position
+    $mdc-text-field-trailing-icon-position,
+    $query: $query
   );
 
   .mdc-text-field__input {
-    padding-right: $mdc-text-field-icon-padding;
-    padding-left: $mdc-text-field-icon-padding;
+    @include mdc-feature-targets($feat-structure) {
+      padding-right: $mdc-text-field-icon-padding;
+      padding-left: $mdc-text-field-icon-padding;
+    }
   }
 }
 
-@mixin mdc-text-field-dense-with-both-icons_ {
+@mixin mdc-text-field-dense-with-both-icons_($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
   @include mdc-text-field-icon-horizontal-position-two-icons_(
     $mdc-text-field-dense-icon-position,
-    $mdc-text-field-dense-icon-position
+    $mdc-text-field-dense-icon-position,
+    $query: $query
   );
 
   .mdc-text-field__input {
-    padding-right: $mdc-text-field-dense-icon-padding;
-    padding-left: $mdc-text-field-dense-icon-padding;
+    @include mdc-feature-targets($feat-structure) {
+      padding-right: $mdc-text-field-dense-icon-padding;
+      padding-left: $mdc-text-field-dense-icon-padding;
+    }
   }
 }
 
 // Full Width
-@mixin mdc-text-field-fullwidth_ {
-  @include mdc-text-field-bottom-line-color($mdc-text-field-bottom-line-idle);
-  @include mdc-text-field-disabled-bottom-line-color($mdc-text-field-bottom-line-idle);
+@mixin mdc-text-field-fullwidth_($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
 
-  width: 100%;
+  @include mdc-text-field-bottom-line-color($mdc-text-field-bottom-line-idle, $query: $query);
+  @include mdc-text-field-disabled-bottom-line-color($mdc-text-field-bottom-line-idle, $query: $query);
+
+  @include mdc-feature-targets($feat-structure) {
+    width: 100%;
+  }
 
   &:not(.mdc-text-field--textarea) {
-    @include mdc-states-base-color(transparent);
-    @include mdc-text-field-fill-color(transparent);
+    @include mdc-states-base-color(transparent, $query: $query);
+    @include mdc-text-field-fill-color(transparent, $query: $query);
 
-    display: block;
+    @include mdc-feature-targets($feat-structure) {
+      display: block;
+    }
 
     .mdc-text-field__input {
-      padding: 0;
+      @include mdc-feature-targets($feat-structure) {
+        padding: 0;
+      }
     }
   }
 
   &.mdc-text-field--textarea .mdc-text-field__input {
-    resize: vertical;
+    @include mdc-feature-targets($feat-structure) {
+      resize: vertical;
+    }
   }
 }
 
-@mixin mdc-text-field-fullwidth-invalid_ {
-  @include mdc-text-field-bottom-line-color($mdc-text-field-error);
+@mixin mdc-text-field-fullwidth-invalid_($query: mdc-feature-all()) {
+  @include mdc-text-field-bottom-line-color($mdc-text-field-error, $query: $query);
 }
 
 // Textarea
-@mixin mdc-text-field-textarea-disabled_ {
-  @include mdc-text-field-outlined-disabled_;
-  /* @alternate */
```

Comment:
Revert `/* @alternate */`. This is required to prevent failure with duplicate CSS property check.